config/config_test.go

```diff
@@ -28,9 +30,35 @@ func Test_BufferHttpModeDefaultsToFalse(t *testing.T) {
 	}
 }
 
+func Test_UpstreamURL(t *testing.T) {
+	urlVal := "http://127.0.0.1:8082"
+	env := []string{
+		fmt.Sprintf("upstream_url=%s", urlVal),
+	}
+
+	actual := New(env)
+	want := urlVal
+	if actual.UpstreamURL != want {
+		t.Errorf("Want %v. got: %v", want, actual.UpstreamURL)
+	}
+}
+
+func Test_UpstreamURLVerbose(t *testing.T) {
+	urlVal := "http://127.0.0.1:8082"
+	env := []string{
+		fmt.Sprintf("http_upstream_url=%s", urlVal),
+	}
+
+	actual := New(env)
+	want := urlVal
+	if actual.UpstreamURL != want {
+		t.Errorf("Want %v. got: %v", want, actual.UpstreamURL)
+	}
+}
+
```

Comment:
Not sure if setenv/unsetenv is a good approach with unit testing.
I suppose those variables are loaded from environment variables. Also what would happen if both variables are set with different values. Should there be any prioritization?

Reply:
That's the reason that those methods are not used. Instead an immutable, stateless slice is used (`env`)